network/p2p/libp2pNode.go

```diff
@@ -55,130 +51,84 @@ type NodeAddress struct {
 // Node is a wrapper around LibP2P host.
 type Node struct {
 	sync.Mutex
-	name                 string                          // friendly human readable Name of the node
-	libP2PHost           host.Host                       // reference to the libp2p host (https://godoc.org/github.com/libp2p/go-libp2p-core/host)
-	logger               zerolog.Logger                  // for logging
-	ps                   *pubsub.PubSub                  // the reference to the pubsub instance
+	libP2PHost           *libP2PHostWrapper              // used to provide unicast and pubsub services
+	logger               zerolog.Logger                  // used to provide logging
 	topics               map[string]*pubsub.Topic        // map of a topic string to an actual topic instance
 	subs                 map[string]*pubsub.Subscription // map of a topic string to an actual subscription
-	conMgr               ConnManager                     // the connection manager passed in to libp2p
-	connGater            *connGater                      // the connection gator passed in to libp2p
+	nodeAddress          NodeAddress                     // used to encapsulate identity of node at network layer
 	flowLibP2PProtocolID protocol.ID                     // the unique protocol ID
 }
 
-// Start starts a libp2p node on the given address.
-func (n *Node) Start(ctx context.Context,
-	nodeAddress NodeAddress,
+func NewLibP2PNode(ctx context.Context,
```

Comment:
Changed `Start` method to `NewLibP2PNode` factory. The `Start` method was acting as the factory of `Node`, however, behaviorally, it does not need a start. Since libp2p host is in a ready state when it is created. Also, we aim at the passing node as a dependency on middleware, so it should be at the ready state by the time it is injected.